pkg/middleware/http/http.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"sync"
+
+	compapi "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	"github.com/dapr/dapr/pkg/config"
+	"github.com/dapr/dapr/pkg/middleware"
+	"github.com/dapr/dapr/pkg/middleware/store"
+	"github.com/dapr/kit/logger"
+)
+
+var log = logger.NewLogger("dapr.middleware.http")
+
+// HTTP returns HTTP middleware pipelines. These pipelines dynamically update
+// the middleware chain when a component is added or removed from the store.
+// Callers need only build a Pipeline once for a given spec.
+type HTTP struct {
```

Comment:
```suggestion
type Chains struct {
```

Something not like HTTP is good to me.